### PR TITLE
feat: add support for noImplicitThis

### PR DESCRIPTION
**Pull request by crisbeto**
Adds support for the `noImplicitThis` compiler option and fixes the compilation errors.

### CODE REVIEW

File: src/cdk/a11y/interactivity-checker/interactivity-checker.spec.ts [hunk 1]
```diff
@@ -509,7 +509,7 @@ describe('InteractivityChecker', () => {
     }
   }
 
-  function runIf(condition: boolean, runFn: Function): () => void {
+  function runIf(this: any, condition: boolean, runFn: Function): () => void {
```
Comment (DevVersion) [inline src/cdk/a11y/interactivity-checker/interactivity-checker.spec.ts]: For a second, I thought that this is an actual parameter.

**Comment by crisbeto** [comment]
This should be the last one missing from the `strict` flag. I talked to @jelbourn about it, but we decided not to use `strict`, because things can start breaking if the TS team adds another check to it.

**Comment by DevVersion** [comment]
@crisbeto Yeah that makes sense. I assumed they would only add these in major releases (where we can expect such breaking changes)

**Comment by josephperrott** [comment]
@crisbeto Looks like the e2e tests are failing on this now?

I was able to reproduce the failure locally, though at a glance am unsure of the cause.

**Comment by crisbeto** [comment]
Fixed @josephperrott.

**Pull request merged**